packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1367,7 +1368,20 @@ class _NestedScrollPosition extends ScrollPosition implements ScrollActivityDele
 
   @override
   void restoreScrollOffset() {
-    if (coordinator.canScrollBody)
+    // If the children of a NestedScrollView have PageStorageKey, the
+    // ScrollOffset should be restored correctly and not controlled by
+    // coordinator.canScrollBody.
+    // Related issues: https://github.com/flutter/flutter/issues/40740
```

Comment:
we shouldn't link to issues in comments, just include all the relevant information in the code

Reply:
Now updated.